chia/full_node/mempool_check_conditions.py

```diff
@@ -97,20 +118,51 @@ def get_name_puzzle_conditions(
 
     flags = MEMPOOL_MODE if mempool_mode else 0
     try:
-        err, result, clvm_cost = GENERATOR_MOD.run_as_generator(max_cost, flags, block_program, block_program_args)
+        err, result = GENERATOR_MOD.run_as_generator(max_cost, flags, block_program, block_program_args)
+
         if err is not None:
+            assert err != 0
             return NPCResult(uint16(err), [], uint64(0))
-        else:
-            npc_list = []
-            for r in result:
-                conditions = []
-                for c in r.conditions:
-                    cwa = []
-                    for cond_list in c[1]:
-                        cwa.append(ConditionWithArgs(ConditionOpcode(bytes([cond_list.opcode])), cond_list.vars))
-                    conditions.append((ConditionOpcode(bytes([c[0]])), cwa))
-                npc_list.append(NPC(r.coin_name, r.puzzle_hash, conditions))
-            return NPCResult(None, npc_list, uint64(clvm_cost))
+
+        condition_cost = 0
+        first = True
+        npc_list = []
+        for r in result.spends:
+            conditions: Dict[ConditionOpcode, List[ConditionWithArgs]] = {}
+            if r.height_relative is not None:
+                add_int_cond(conditions, ConditionOpcode.ASSERT_HEIGHT_RELATIVE, r.height_relative)
+            if r.seconds_relative > 0:
+                add_int_cond(conditions, ConditionOpcode.ASSERT_SECONDS_RELATIVE, r.seconds_relative)
+            for cc in r.create_coin:
+                if cc[2] == b"":
```

Comment:
this is a slight difference. the new clvm_rs API always returns a 3-tuple with the last element being `bytes`. If there is no hint, the `bytes` is empty. However, previously, an empty atom was a valid hint, distinct from no hint at all.

Since hints aren't consensus critical, I don't think this really matters, but it meant I had to change the `test_rom.py` below.